lib/content_publisher/document_collection_group_membership_migrator.rb

```diff
@@ -0,0 +1,28 @@
+module ContentPublisher
+  class DocumentCollectionGroupMembershipMigrator
+    attr_reader :document
+
+    def initialize(document)
+      @document = document
+    end
+
+    def call
+      edition = document.published_edition.present? ? document.published_edition : document.latest_edition
+
+      document_collection_group_memberships = DocumentCollectionGroupMembership.where(document_id: document.id)
+
+      return unless document_collection_group_memberships.any?
```

Comment:
```suggestion
      return unless DocumentCollectionGroupMembership.exists?(document_id: document.id)
```

You can probably lose the check via variable and call an exists directly on the model class.